app/src/main/kotlin/com/simplemobiletools/gallery/pro/activities/ViewPagerActivity.kt

```diff
@@ -388,7 +371,17 @@ class ViewPagerActivity : SimpleActivity(), ViewPager.OnPageChangeListener, View
             }
         }
 
-        refreshViewPager()
+        // show the selected image asap, while loading the rest in the background to allow swiping between them. Might be needed at third party intents
+        if (mMediaFiles.isEmpty() && mPath.isNotEmpty() && mDirectory != FAVORITES) {
```

Comment:
you basically just removed the cache file displaying, while leaving the no longer valid comment there... That code was there for a reason, now it can take a long time for anything to be shown.

Reply:
line 374 is a valid comment, read lines 374 to 384. `refreshViewPager()` does the background loading instead.

The file refresh done in `refreshViewPager()` almost always executes the callback a little before `getCachedMedia()` does the db lookup. Confirmed with 100s of files a couple of times on a couple of devices. Sometimes the path specified in the external intent might not be cached in the db and `getCachedMedia()` might return empty or incomplete data and cause the viewpager to snap to some invalid position

The `getCachedMedia()` was introduced in PR https://github.com/SimpleMobileTools/Simple-Gallery/commit/318664fca1b751dc6f14140b3d0c6cf91fa934f7 for external intents as a fix for https://github.com/SimpleMobileTools/Simple-Gallery/issues/2332 (files were shown after the loading was complete thus delayed)...

> now it can take a long time for anything to be shown

... so that still works as per my testing and as per lines 374-382.

Reply:
well ok, that hackfix has been added for the previous way of fetching files, lets see what will the user report back at th given issue